cmake/compiler/gcc/target_arm.cmake

```diff
@@ -59,8 +59,11 @@ set(LLEXT_APPEND_FLAGS
   -mthumb
 )
 
-set(LLEXT_CFLAGS
-    -mlong-calls
-    -mthumb
+list(APPEND LLEXT_EDK_REMOVE_FLAGS
+    --sysroot=.*
+    -fmacro-prefix-map=.*
+    )
+
+list(APPEND LLEXT_EDK_APPEND_FLAGS
```

Comment:
`LLEXT_*` settings has been misplaced in earlier PR, but let's not continue bad structure of introducing llext_edk_*` settings.

Reply:
I understand your line of thought, but the toolchain infrastructure, and thus the `cmake/compiler/gcc/*` is for setting up the toolchain for building Zephyr. Something which is done in the current CMake invocation.

Where those flags are for exporting when building the extension, which is something quite different.

There is no obvious place for an alternative location atm.
I did consider if llext should have its own CMake module, but that collides a bit with the fact that a llext-edk target in now created.

Accepted to keep for now, but could you file an enhancement issue so that we don't forget a later cleanup, as well as providing a bit of history as to why this location was accepted. So that we can reference in future PRs.